tests/FSharp.Compiler.Private.Scripting.UnitTests/FSharpScriptTests.fs

```diff
@@ -87,22 +89,20 @@ x
 
     [<Fact>]
     member _.``Capture console input``() =
-        use script = new FSharpScript(input = "stdin:1234\r\n")
+        use _ = new TestConsole.ProvideInput("stdin:1234\r\n")
+        use script = new FSharpScript()
         let opt = script.Eval("System.Console.ReadLine()") |> getValue
         let value = opt.Value
         Assert.Equal(typeof<string>, value.ReflectionType)
         Assert.Equal("stdin:1234", downcast value.ReflectionValue)
 
     [<Fact>]
     member _.``Capture console output/error``() =
+        use capture = new TestConsole.ExecutionCapture()
```

Comment:
Am I right with seeing this test as the main demonstration of the proposed test-API changes?
If yes, I like it :)